Azure/azure-iot-cli-extension
# Storage test updates

Test runs:
https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=6010&view=results
https://dev.azure.com/azureiotdevxp/aziotcli/_build/results?buildId=6012&view=results

Slightly smarter way to run the storage tests (upload file and export/import devices)
Things to note:
- one export or import device job per iot hub, no matter the sku
- add polling checks to the export and import device jobs
- add while loop/retry for all export or import device job tests: re-add permissions, retry the export and import tests without creating a new iot hub. 
- add a check for export and import device active job before rerunning the storage tests with sas token

Tests can still fail but they fail less often:
- if there is a 403001 - it is most likely that there is a job running at the same time. The different checks are there to make sure we do not run into this.
- if there is a 400007 - the export job failed and the import job failed as a result (no file to import from). This or should guard against this by failing the test if a job fails (and if there are still retries, retry)
- note that the export and import jobs may fail or take up to 4 hours randomly due to the nature of a distributed system. 

---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

Intent for Production

- [ ] It is expected that pull requests made to default or core branches such as `dev` or `main` are of production grade. Corollary to this, any merged contributions to these branches may be deployed in a public release at any given time. By checking this box, you agree and commit to the expected production quality of code.

Basic expectations

- [ ] If introducing new functionality or modified behavior, are they backed by unit and/or integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** the relevant unit **and** integration tests pass? i.e. `pytest <project root> -vv`. Please provide evidence in the form of a screenshot showing a succesful run of tests locally OR a link to a test pipeline that has been run against the change-set.
- [ ] Have linter checks passed using the `.pylintrc` and `.flake8` rules? Look at the CI scripts for example usage.
- [ ] Have extraneous print or debug statements, commented out code-blocks or code-statements (if any) been removed from the surface area of changes?
- [ ] Have you made an entry in HISTORY.rst which concisely explains your user-facing feature or change?

Azure IoT CLI maintainers reserve the right to enforce any of the outlined expectations.

A PR is considered **ready for review** when all basic expectations have been met (or do not apply).
